src/odin/maneuversbuilder.cc

```diff
@@ -367,6 +368,37 @@ void ManeuversBuilder::Combine(std::list<Maneuver>& maneuvers) {
         curr_man = next_man;
         ++next_man;
       }
+      // Combine double L turns or double R turns in short non-internal intersections as u-turns
+      else if (!curr_man->internal_intersection() &&
+               curr_man->travel_mode() == TripLeg_TravelMode::TripLeg_TravelMode_kDrive &&
+               curr_man->length(Options::kilometers) <= (kMaxInternalLength * kKmPerMeter) &&
+               prev_man->HasSimilarNames(&(*next_man), true) && curr_man != next_man &&
+               !next_man->IsDestinationType()) {
+        if ((curr_man->type() == DirectionsLeg_Maneuver_Type_kLeft ||
+             curr_man->begin_relative_direction() == Maneuver::RelativeDirection::kLeft) &&
+            (next_man->type() == DirectionsLeg_Maneuver_Type_kLeft ||
+             next_man->begin_relative_direction() == Maneuver::RelativeDirection::kLeft)) {
+          curr_man->set_type(DirectionsLeg_Maneuver_Type_kUturnLeft);
+          maneuvers_have_been_combined = true;
+          LOG_TRACE(
+              "+++ Combine: double L turns in short non-internal intersection as ManeuverType=SIMPLE_UTURN_LEFT +++");
+        } else if ((curr_man->type() == DirectionsLeg_Maneuver_Type_kRight ||
+                    curr_man->begin_relative_direction() == Maneuver::RelativeDirection::kRight) &&
+                   (next_man->type() == DirectionsLeg_Maneuver_Type_kRight ||
+                    next_man->begin_relative_direction() == Maneuver::RelativeDirection::kRight)) {
+          curr_man->set_type(DirectionsLeg_Maneuver_Type_kUturnRight);
+          maneuvers_have_been_combined = true;
+          LOG_TRACE(
+              "+++ Combine: double R turns in short non-internal intersection as ManeuverType=SIMPLE_UTURN_RIGHT +++");
+        }
+        if (maneuvers_have_been_combined)
+          curr_man =
+              CombineDoubleTurnToUturnManeuver(maneuvers, prev_man, curr_man, next_man, is_first_man);
+        if (is_first_man) {
```

Comment:
can not combine if first man - need to check at beginning of block

Reply:
therefore, do not need to `is_first_man` pass to CombineDoubleTurnToUturnManeuver